src/main/java/com/exasol/dbbuilder/dialects/postgres/PostgreSqlSchema.java

```diff
@@ -0,0 +1,33 @@
+package com.exasol.dbbuilder.dialects.postgres;
+
+import com.exasol.db.Identifier;
+import com.exasol.dbbuilder.dialects.*;
+
+/**
+ * PostgreSQL {@link Schema}.
+ */
+public class PostgreSqlSchema extends AbstractSchema {
+    private final PostgreSqlImmediateDatabaseObjectWriter writer;
+
+    /**
+     * Create a new instance of {@link PostgreSqlSchema}.
+     * 
+     * @param writer object writer
+     * @param name   name of the schema
+     */
+    public PostgreSqlSchema(final PostgreSqlImmediateDatabaseObjectWriter writer, final String name) {
+        super(PostgreSqlIdentifier.of(name));
+        this.writer = writer;
+        this.writer.write(this);
```

Comment:
I find this dangerous, as we are calling some behavior on an object that is not yet fully initialized.

Reply:
I'll create a ticket for it